kernel/rtlil.h

```diff
@@ -842,20 +842,25 @@ struct RTLIL::SigSpec
 {
 private:
 	int width_;
+	bool packed_;
 	unsigned long hash_;
-	std::vector<RTLIL::SigChunk> chunks_; // LSB at index 0
-	std::vector<RTLIL::SigBit> bits_; // LSB at index 0
+	union {
+		std::vector<RTLIL::SigChunk> chunks_; // LSB at index 0
+		std::vector<RTLIL::SigBit> bits_; // LSB at index 0
```

Comment:
Could we do this with `std::variant`?

Reply:
Good idea, this is a good small but hot case to try it on